src/ResourceManager/Common/Commands.ResourceManager.Common/AzureRMCmdlet.cs

```diff
@@ -214,6 +219,11 @@ protected override void InitializeQosEvent()
                         s => string.Format(CultureInfo.InvariantCulture, "-{0} ***", s)));
             }
 
+            if (!string.IsNullOrWhiteSpace(this.ParameterSetName))
```

Comment:
@praries880 since we don't need to do any additional checks (like we do with `this.MyInvocation` in the if-statements above), is there any reason we can't move the initialization of this property into the instantiation of the `AzurePSQoSEvent` object above?

```cs
_qosEvent = new AzurePSQoSEvent()
{
    CommandName = commandAlias,
    ...
    ParameterSetName = this.ParameterSetName
}
```

Also, I don't believe `ParameterSetName` can ever be null or empty (unless the cmdlet creator gives an empty set name, which doesn't happen in our cmdlets 😀 )

Reply:
didnt know that either ;)